Topic 2: Test Driven Development

hherminioandres opened the pull request:
1 - Use TDD to create the 'Recent file list' behaviour. Some examples of this behaviour is:
-When the program is run for the first time, the list is empty.
-When a file is opened, it is added to the recent file list.
-If an opened file already exists in the recent file list, it is bumped to the top, not duplicated in the list.
-If the recent file list gets full (typical number of items is 15), the oldest item is removed when a new item is added.
2 - Using TDD techniques such as mocking, faking and stubs, try to develop your own blog software featuring:
-Post new entry
-Delete existing entry
-Show 10 most recent entries

hherminioandres committed:
Recent files list

Create the recent file list behaviour

hherminioandres committed:
Blog - TDD

Created blog using TDD techniques:  mocking, faking and stubs.

hherminioandres committed:
Correction - Tabulation

Tabulation error was fixed

salaesp commented on src/test/java/com/blog/BlogTest.java:
```diff
@@ -0,0 +1,57 @@
+	public void getPostMock() {
+		Post post = null; 
+		IBlog blogMock = mock(IBlog.class);
+		when(blogMock.getPost(NAMEP)).thenReturn(post);
```
since you are testing the blog, you shouldnt mock it

salaesp commented on src/test/java/com/blog/BlogTest.java:
```diff
@@ -0,0 +1,57 @@
+		FakeBlog fakeBlog = new FakeBlog();
+		fakeBlog.setName(NAMEB);
+		assertEquals(fakeBlog.getName(), NAMEB);
+	}
```
you are not testing getting the last X posts.

hherminioandres committed:
Topic 2: Several corrections

-Removed the IBlog interface because not use.
-Added the PostDB class. It allows to posts can be add, delete or search.
-Now "mock" is correct implemented.

hherminioandres committed:
Orderig files
